### ISSUE TITLE
Network Configuration Using cloud-init

### ISSUE THREAD

**Issue by j-petit** [hetznercloud/cli#297]
Hi,
I am trying to do network interface configuration via `cloud-init` and `hcloud` like
```
hcloud server create --name private-client-3 --type cx11 --image ubuntu-20.04 --user-data-from-file test.yml
```
where the `test.yml` is
```
#cloud-config

network:
  version: 2
  ethernets:
    ens10:
      dhcp4: true
      gateway4: 10.253.0.1
      nameservers:
        addresses: [8.8.8.8]

final_message: "The system is finally up"
```
However, the system seems not to be configured correctly - the content of `/etc/netplan/50-cloud-init.yaml` is still
```
network:
    version: 2
    ethernets:
        eth0:
            addresses:
            - 2a01:4f8:1c1c:2d81::1/64
            dhcp4: true
            gateway6: fe80::1
            match:
                macaddress: 96:00:00:96:1c:33
            set-name: eth0
```
which does not correspond to my desired configuration. Do you have any ideas what could be the problem that my user-data is not applied? `cloud-init status --long` reports no errors
```
status: done
time: Thu, 11 Feb 2021 09:20:46 +0000
detail:
DataSourceHetzner
```

**Comment by LKaemmerling** [comment]
Hey @j-petit, you should disable `dhcp4` on ens10 otherwise it would overwrite it always. But I wouldn't recommend that. I guess your plan is to send the whole traffic of this VM over the private interface to the server `10.253.0.1` or?

**Comment by j-petit** [comment]
Yes, the idea is that all traffic gets routed over a firewall. Manually setting up the interfaces via netplan works fine, the problem is the configuration via `cloud-init` using the `--user-data-from-file` input. 

I did try with `dhcp4: false` however this does not really change anything. `/etc/netplan/50-cloud-init.yaml` stays the same.

**Comment by ksbbf** [comment]
@j-petit well, some time ago ...

Did you find any solution for your problem? I have exactly the same issue. nothing in the cloud-init.yaml is processed.

**Comment by j-petit** [comment]
Hi, no we have moved to another cloud provider but for different reasons.

**Comment by ksbbf** [comment]
Dear all,

the question here is not the network-configuration. It's only a sample for user-data.  

The question is why cloud-init does simply nothing: meanwhile I figured out that there seems to be a caching problem or similar.

I started anew: 
1. I created a server, maybe without --user-data.
2. I deleted it.
3. I created it again with user-data. (The cloud-init-file was tested by a colleague.): the cloud-init-file was **not** processed.
4.  I tried several times. No succcess.
5. The debug-log said "The name of the server ist already used", although it was deleted.
6. I changed the server type, e.g. from cax11 to cax21 and changed the name. Success: cloud-init-file was processed.
7. I deleted it again, changed user-data a little bit. Processed again.
8. I changed the server type (cax21 to cax11) and name again: success.

So I guess there is something wrong with a cache (?) of server configurations. From my perspective the hetzner cloud doesn't seem to delete correctly or reuses configurations with the same name.

So my advice is: change server name and type, check cloud-innit-execution. if necessary switch back to older configuration.

By the way: There is a good sample for network configuration in the docs <https://community.hetzner.com/tutorials/how-to-set-up-nat-for-cloud-networks#step-6---cloud-init>
